### DISCUSSION TITLE
Passing 'nonce' to Head tag generate noscript tag with nonce value accessible by JavaScript, Is that security risk?

### DISCUSSION THREAD

**Discussion by S-YOU** [vercel/next.js#49220] [Help]
### Summary

I am passing nonce in _document.tsx like this

```
<Html...>
<Head nonce={nonce}/>
...
</Html>
```

Which generate like this

![image](https://user-images.githubusercontent.com/1001175/236253398-db61c80c-d6f1-4cb8-9418-bf601aaddb5f.png)

I can access it from JavaScript like this

![image](https://user-images.githubusercontent.com/1001175/236253823-d5240d86-c8d8-4d76-8aa2-9797149b9b3b.png)

Is this security risk?

**Comment by leerob** [comment]
Hey folks, wanted to swing back here with an update. After digging through many different issues and discussions, I've [**made a new page in the documentation**](https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy) ([PR](https://github.com/vercel/next.js/pull/54601)) specifically for Content Security Policy and nonces. This docs page: 

- Explains how to generate a `nonce` with Middleware
- Shows how to consume the `nonce` in a route with `headers()`
- Shows a complete CSP without needing to use any `unsafe`
- Shows how to ignore the `nonce` Middleware from running on prefetches / static assets

Further, we've patched some bugs and made improvements to `nonce` handling in Next.js that will be available in the latest `canary` version (for those of you time traveling from the future, upgrade to Next.js 13.5). We also updated the `with-strict-csp` example in the `examples/` folder, which is backlinked from the [new documentation page](https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy).

Really hope this helps out, thank you all 🙏 I'll be closing this discussion out. To continue the discussion, [please go here](https://github.com/vercel/next.js/discussions/54907).
